cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java

```diff
@@ -1273,6 +1276,76 @@ public void testUpdateSchedulesFlag() throws Exception {
     Assert.assertEquals(2, actualSchedules.size());
   }
 
+  @Test
+  public void testAddScheduleWithDisabledProfile() throws Exception {
+    // put my profile and disable it
+    ProfileId profileId = new NamespaceId(TEST_NAMESPACE1).profile("MyProfile");
+    putProfile(profileId, Profile.DEFAULT, 200);
+    disableProfile(profileId, 200);
+
+    // deploy an app with schedule, the deploy should success since the schedules contain no profile info
+    AppWithSchedule.AppConfig config = new AppWithSchedule.AppConfig(true, true, true);
+
+    Id.Artifact artifactId = Id.Artifact.from(Id.Namespace.fromEntityId(TEST_NAMESPACE_META1.getNamespaceId()),
+      AppWithSchedule.NAME, VERSION1);
+    addAppArtifact(artifactId, AppWithSchedule.class);
+    AppRequest<? extends Config> request = new AppRequest<>(
+      new ArtifactSummary(artifactId.getName(), artifactId.getVersion().getVersion()), config, null, null, false);
+
+    ApplicationId defaultAppId = TEST_NAMESPACE_META1.getNamespaceId().app(AppWithSchedule.NAME);
+    Assert.assertEquals(200, deploy(defaultAppId, request).getStatusLine().getStatusCode());
+
+    ScheduleProgramInfo programInfo = new ScheduleProgramInfo(SchedulableProgramType.WORKFLOW,
+      AppWithSchedule.WORKFLOW_NAME);
+
+    // in the schedule property put my disabled profile
+    ImmutableMap<String, String> properties = ImmutableMap.of(SystemArguments.PROFILE_NAME, "USER:MyProfile");
+    TimeTrigger timeTrigger = new TimeTrigger("0 * * * ?");
+    ScheduleDetail timeDetail = new ScheduleDetail(TEST_NAMESPACE1,
+      AppWithSchedule.NAME, ApplicationId.DEFAULT_VERSION,
+      "FailSchedule", "Add should fail for me", programInfo, properties,
+      timeTrigger, Collections.<Constraint>emptyList(),
+      Schedulers.JOB_QUEUE_TIMEOUT_MILLIS, null);
+
+    // it should fail because of the disabled profile
+    HttpResponse response = addSchedule(TEST_NAMESPACE1, AppWithSchedule.NAME, null, "FailSchedule", timeDetail);
+    Assert.assertEquals(HttpResponseStatus.CONFLICT.code(), response.getStatusLine().getStatusCode());
+    Assert.assertTrue(EntityUtils.toString(response.getEntity()).contains(
+      String.format("Profile %s in namespace %s is disabled", profileId.getProfile(), profileId.getNamespace())));
+  }
+
+  @Test
+  public void testStartProgramWithDisabledRuntimeArgs() throws Exception {
+    // We will use default profile now for testing since we treat it as on premise and all other profiles as isolated
+    // See ProgramLifeCycleService runInternal() method for more information
+    disableProfile(ProfileId.DEFAULT, 200);
+
+    // deploy, check the status
+    deploy(WordCountApp.class, 200, Constants.Gateway.API_VERSION_3_TOKEN,
+           TEST_NAMESPACE1);
+
+    Id.Flow wordcountFlow1 = Id.Flow.from(TEST_NAMESPACE1, WORDCOUNT_APP_NAME, WORDCOUNT_FLOW_NAME);
```

Comment:
we are going to restrict profiles to workflows and schedules in 5.0.

Reply:
Changed to test on a workflow